.changeset/tiny-days-dance.md

```diff
@@ -0,0 +1,26 @@
+---
+'astro': minor
+---
+
+Cookie encoding / decoding can now be customized
+
+Adds new `encode` and `decode` functions to allow customizing how cookies are encoded and decoded. For example, you can bypass the default encoding via `encodeURIComponent` when adding a URL as part of a cookie:
+
+```astro
+---
+Astro.cookies.set('url', Astro.url.toString(), {
+  // Override the default encoding so that URI components are not encoded
+  encode: o => o
+});
+---
```

Comment:
```suggestion
---
import { encodeCookieValue } from "./cookies";
Astro.cookies.set('url', Astro.url.toString(), {
  // Override the default encoding so that URI components are not encoded
  encode: value => encodeCookieValue(value)
});
---
```

Reply:
I like Ema's changes to the code sample!

Reply:
Yes! Thanks so much, i was struggling with this, as this is *really* the use-case in mind, so just giving it a function name does help a lot.